examples/colors.rs

```diff
@@ -107,7 +106,7 @@ fn render_fg_named_colors(frame: &mut Frame, bg: Color, area: Rect) {
                 .split(*area)
                 .to_vec()
         })
-        .collect_vec();
+        .collect::<Vec<_>>();
     for (i, &fg) in NAMED_COLORS.iter().enumerate() {
         let color_name = fg.to_string();
         let paragraph = Paragraph::new(color_name).fg(fg).bg(bg);
```

Comment:
this block (can't highlight it fully in the review) can more reasonably written as:
```rust
    let vertical = Layout::vertical([Constraint::Length(1); 2]).split(inner);
    let areas = vertical.iter().flat_map(|area| {
        Layout::horizontal([Constraint::Ratio(1, 8); 8])
            .split(*area)
            .to_vec()
    });
    for (bg, area) in zip(NAMED_COLORS, areas) {
        let color_name = bg.to_string();
        let paragraph = Paragraph::new(color_name).fg(fg).bg(bg);
        frame.render_widget(paragraph, area);
    }
```
Which avoids the allocation and itertools better than the `collect::<Vec<_>>()` code.

This is the same with many of the changes to using that (I haven't checked them all).

This is a more complex changes however. Would you be willing to make it?